ouroboros-consensus-mock/src/Ouroboros/Consensus/Mock/Protocol/Praos.hs

```diff
@@ -78,31 +78,97 @@ import           Ouroboros.Consensus.Protocol.Abstract
 import           Ouroboros.Consensus.Protocol.Signed
 import           Ouroboros.Consensus.Util.Condense
 
+-- The Praos paper can be located at https://ia.cr/2017/573
+--
+-- A block as defined in Praos (section 3, definition 2, then extended in Fig 9)
+-- consist of a tuple:
+--
+-- > B = (slⱼ, st, d, B_πj, ρ, σⱼ)
+--
+-- where
+-- - slⱼ: the slot at which the block was generated.
+--
+-- - st: the current state.
+--
+-- - d: data (transaction data in most cases).
+--
+-- - B_πj: block proof consisting of (Uᵢ, y, π).
+--      - y: a VRF output used to confirm tha Uᵢ was the slot leader.
+--      - π: the VRF proof of the above value.
+--
+--      > (y, π) ≜ VRF_evalProve(η, slⱼ, TEST) see Fig 9
+--
+-- - ρ: the block nonce consisting of (ρ_y, ρ_π), to capture entropy from the
+--      block forging process.
+--      - ρ_y: a VRF output used to confirm this block captured all the previous
+--             entropy.
+--      - ρ_π: the VRF proof of the above value.
+--
+--      > (ρ_y, ρ_π) ≜ VRF_evalProve(η, slⱼ, NONCE) see Fig 9
+--
+-- - σⱼ: a signature on (st, d, slⱼ, B_πj, ρ) with the signing key
+--       for the slot slⱼ for the stakeholder Uᵢ.
+--
+-- Protocol parameters:
+-- - k: maximum number of blocks we can rollback.
+-- - R: number of slots per epoch.
+-- - f: the active slots coefficient, specifies roughly the proportion of
+--      occupied slots per epoch.
+--
+-- Some values you will encounter:
+-- - η: The epoch's nonce. Captures entropy from the block chain. See Fig 8 in
+--      praos paper for where it is used and Fig 10 for how it is defined.
+--      Defined as the hash of the η from the previous epoch, this epoch number
+--      and the ρ of the first 2/3 of the blocks in the previous epoch.
+--
+--      > η_e ≜ HASH(η_{e-1} || e || ρ_{e-1,0} ... ρ_{e-1, 2R/3})
+--
+-- - Tᵢ: the leader threshold for a specific stakeholder considering its
+--       relative stake (therefore depending on the slot). Defined in the Praos
+--       paper in Figure 4 using the definition for ϕ_f(αᵢ) from section 3.3.
+--
+--       > Tᵢ ≜ 2^(l_VRF) * pᵢ
+--       > pᵢ = ϕ_f(αᵢ) ≜ 1 - (1 - f)^(αᵢ)
+
 {-------------------------------------------------------------------------------
-  Fields required by Praos in the header
+  Fields required by Praos in the header of the block
```

Comment:
Not a problem, but beware that "header" implicitly means "header of a block" throughout the code base, unless specified otherwise (eg in some lower-level Network module, likely).

Reply:
Removed, I used to deal with both Block Header and Transaction Header, that’s why I wrote that explicit description, but it seems redundant now.